### DISCUSSION TITLE
Minimum memory requirements for running cFS with an RTOS

### DISCUSSION THREAD

**Discussion by rvy11** [nasa/cFS#781] [Q&A]
Hello,

My team and I are interested in evaluating flight software on a high-performance MCU and we are looking at the [TMS570LC457](https://www.ti.com/product/TMS570LC4357) which was used by the _Ingenuity_ Mars helicopter. The specs include:

CPU - 300-MHz
SRAM: 512KB, SDRAM: 8MB
Code storage: 4MB Flash

Have there been previous efforts to run cFS baremetal or with an RTOS like RTEMS on a platform with similar specs as above? Are there minimum memory requirements for running cFE on a particular target?

**Comment by skliper** [comment]
See table 1 (pg 12) for projects and hardware (where known): https://ntrs.nasa.gov/api/citations/20205010026/downloads/CR%2020205010026%20Core%20Flight%20System%20(cFS)%20Flight%20Software%20(FSW)%20Final%20Report-1.pdf

Depends a bit on how much margin and how many copies (if any) that you need in storage to meet your mission needs... but those numbers look in family w/ some of the heritage cFS platforms that I've seen.  I don't know of anyone running cFS bare metal, but RTEMS is pretty common.
